server/channels/app/authorization.go

```diff
@@ -82,6 +82,19 @@ func (a *App) SessionHasPermissionToChannel(c request.CTX, session model.Session
 		return false
 	}
 
+	channel, appErr := a.GetChannel(c, channelID)
+	if appErr != nil && appErr.StatusCode == http.StatusNotFound {
+		return false
+	}
+
+	if session.IsUnrestricted() || session.Roles == model.SystemAdminRoleId {
```

Comment:
Permissions within a role could possible change, so we should check permissions, not roles. Admittedly, with the "admin" role, its a little bit of splitting hairs, since its currently not possible, to remove the `PermissionManageSystem` permission.  But never know about in the future.

```suggestion
	if session.IsUnrestricted() || a.RolesGrantPermission(session.GetUserRoles(), model.PermissionManageSystem.Id) {
```

Reply:
I guess I only fixed the one :P